fix: LSDV-4907: Import - validate filetypes on the FE

jombooth opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



### Describe the reason for change
https://labelstudio.aha.io/develop/comments/7247239546298336794

We have filetype validation on the backend, but we want it on the frontend also to show more helpful error messages to users, and avoid server calls that will 4xx. We could check filetypes in greater depth on the FE but since we also have backend validation I feel that's not necessary and this lightweight approach will suffice.

Also remove `.aiff` and `.au` because per @bmartel these are not supported by the backend

#### What does this fix?
_(if this is a bug fix)_



#### What is the new behavior?
_(if this is a breaking or feature change)_

Attempt to add multiple files, one of which is the wrong type. None will be uploaded and we'll show an error for the first invalid file.
![image](https://github.com/heartexlabs/label-studio/assets/3943358/64a6ba99-ed52-4f93-8760-973b522bc55e)
Files with valid extensions upload as before, and the error state is cleared as expected.
![image](https://github.com/heartexlabs/label-studio/assets/3943358/1fe0efe3-11dd-4add-abf2-3d3e0f427de6)


#### What is the current behavior?
_(if this is a breaking or feature change)_



#### What libraries were added/updated?
_(list all with version changes)_



#### Does this change affect performance?
_(if so describe the impacts positive or negative)_



#### Does this change affect security?
_(if so describe the impacts positive or negative)_



#### What alternative approaches were there?
_(briefly list any if applicable)_



#### What feature flags were used to cover this change?
_(briefly list any if applicable)_



### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit



### Which logical domain(s) does this change affect?
The Import component in the FE.

jombooth committed:
fix: LSDV-4907: Import - validate filetypes on the FE

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/heartexlabs/label-studio/actions/runs/5549397332

jombooth commented on label_studio/frontend/src/pages/CreateProject/Import/Import.js:
```diff
@@ -223,7 +241,13 @@ export const ImportPage = ({
-    for (let f of files) fd.append(f.name, f);
+    for (let f of files) {
+      if (!allSupportedFormats.includes(getFileExtension(f.name))) {
+        onError(new Error(`The filetype of file "${f.name}" is not supported.`))
```
On a cursory glance I didn't see typographic quotes widely used in the repo. question for reviewers - would those be preferred here?

jombooth committed:
use semicolons

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/heartexlabs/label-studio/actions/runs/5549447635

bmartel commented on label_studio/frontend/src/pages/CreateProject/Import/Import.js:
```diff
@@ -13,6 +13,24 @@ function flatten(nested) {
 
+const supportedFormats = {
+  text: ['txt'],
+  audio: ['wav', 'aiff', 'mp3', 'au', 'flac', 'm4a', 'ogg'],
```
```suggestion
  audio: ['wav', 'mp3', 'flac', 'm4a', 'ogg'],
```
These are the only supported audio formats of webaudio. We can just remove the `au` and `aiff`

jombooth committed:
remove aiff and au since those are not supported anymore, FE+BE

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/heartexlabs/label-studio/actions/runs/5582119630

jombooth committed:
add jpEg as on the BE, add note about keeping compatibility

jombooth committed:
Merge branch 'fb-LSDV-4907/frontend-checks-file-extensions' of github.com:heartexlabs/label-studio into fb-LSDV-4907/frontend-checks-file-extensions

jombooth committed:
actually add jpEg

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/heartexlabs/label-studio/actions/runs/5583232992
